cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/StorageManager.java

```diff
@@ -0,0 +1,93 @@
+package com.linkedin.openhouse.cluster.storage;
+
+import static com.linkedin.openhouse.cluster.storage.StorageType.LOCAL;
+
+import com.google.common.base.Preconditions;
+import com.linkedin.openhouse.cluster.storage.configs.StorageProperties;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import javax.annotation.PostConstruct;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * The StorageManager class is responsible for managing the storage types and providing the
+ * appropriate storage implementation based on the configuration.
+ */
+@Component
+public class StorageManager {
+
+  @Autowired StorageProperties storageProperties;
+
+  @Autowired StorageType storageType;
+
+  @Autowired List<Storage> storagesBeans;
+
+  /**
+   * Validate the storage properties.
+   *
+   * <p>It validates storage properties as follows
+   *
+   * <p>1. default-type for storage is not set and storage types are not provided, ie. LocalStorage
+   * is configured **valid**
+   *
+   * <p>2. default-type for storage is set and storage types are provided ,and it contains
+   * default-type **valid**
+   *
+   * <p>all other configurations are **invalid**
+   */
+  @PostConstruct
+  public void validateProperties() {
+    String clusterYamlError = "Cluster yaml is incorrectly configured: ";
+    Preconditions.checkArgument(
+        !(storageProperties.getDefaultType() == null
+            && storageProperties.getTypes() != null
+            && !storageProperties.getTypes().isEmpty()),
+        clusterYamlError + "default-type for storage is not set");
+    Preconditions.checkArgument(
+        !(storageProperties.getDefaultType() != null
+            && (storageProperties.getTypes() == null || storageProperties.getTypes().isEmpty())),
+        "no types are provided");
+    Preconditions.checkArgument(
+        !(storageProperties.getDefaultType() != null
+            && !storageProperties.getTypes().containsKey(storageProperties.getDefaultType())),
+        "default-type was not provided in types");
+    try {
+      Optional.ofNullable(storageProperties.getDefaultType()).ifPresent(storageType::fromString);
+      Optional.ofNullable(storageProperties.getTypes())
+          .map(Map::keySet)
+          .ifPresent(keyset -> keyset.forEach(key -> storageType.fromString(key)));
+    } catch (IllegalArgumentException e) {
+      throw new IllegalArgumentException(clusterYamlError + e.getMessage());
+    }
+  }
+
+  /**
+   * Get the default storage.
+   *
+   * @return the default storage
+   */
+  public Storage getDefaultStorage() {
+    if (storageProperties.getDefaultType() == null) {
+      return getStorage(LOCAL);
```

Comment:
So when local storage is used there is no default type. Do you think we can keep the default type as `local` (at code level) that would simplify things?  As part of deployment default type can be overridden through config with hdfs or other types?

Reply:
the  clusteryaml -> storageProperties parser doesn't support keeping default-type the way ClusterProperties does. The if condition are basically:
1- clusteryaml configured at all
2- clusteryaml properly configured with default-type and types

Reply:
Got it. Thanks for the clarification.